src/task/include/task/TaskClient.hpp

```diff
@@ -60,7 +61,7 @@ class TaskClient
   }
 
   // The client can wait for a result with a timeout
-  TaskStatus waitForResult(typename ResultMsg::SharedPtr result, unsigned int milliseconds)
+  TaskStatus waitForResult(typename ResultMsg::SharedPtr& result, unsigned int milliseconds)
```

Comment:
From the discussion with Carl in our team meeting, we want to use the first version (without the reference).

Reply:
There are some issues with the function without the reference. I found [this](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rr-sharedptrparam).

Reply:
Looking at Carlos' link and thinking some more about it, it seems like the normal usage is to pass as a const &.
Exceptions:
1. You want to share ownership with the callee. Then pass by value. This would apply when the callee is going to hang on to the shared_ptr for some time beyond the length of that one function call.
2. You are passing the shared_ptr as an output parameter. Then pass by reference.

This particular instance looks like exception 2.